### PR TITLE
feat(fluid-popover): Introduce FluidPopover as WebComponent.

### PR DESCRIPTION
**Pull request by peter-affenzeller**
### <strong>Pull Request</strong>

Depends on #1565

More tests will follow...

#### Type of PR

Feature (non-breaking change which adds functionality)

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -2959,6 +2959,53 @@
...
+          }
+        },
+        "test": {
+          "builder": "@nrwl/jest:jest",
+          "options": {
+            "jestConfig": "libs/fluid-elements/popover/jest.config.js",
+            "tsConfig": "libs/fluid-elements/popover/tsconfig.spec.json",
+            "passWithNoTests": true
+          }
+        }
```
Comment (lukasholzer) [inline angular.json:3006]: Please remove the testing target with the cli that should be done with bazel

File: libs/fluid-elements/popover/.stylelintrc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (lukasholzer) [inline libs/fluid-elements/popover/.stylelintrc:1]: this file is not needed please remove

File: libs/fluid-elements/popover/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,10 @@
+load("//tools/bazel_rules:index.bzl", "stylelint")
+
+package(default_visibility = ["//:__subpackages__"])
+
+stylelint(
```
Comment (lukasholzer) [inline libs/fluid-elements/popover/BUILD.bazel:5]: stylelint is not needed as there are no style files. please remove.

Instead add building and testing with bazel

File: libs/fluid-elements/popover/README.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# fluid-popover
+
+This library was generated with [Nx](https://nx.dev).
+
+## Running unit tests
+
+Run `ng test fluid-popover` to execute the unit tests via
```
Comment (lukasholzer) [inline libs/fluid-elements/popover/README.md:7]:
```suggestion
Run `bazel test //libs/barista-components/fluid-popover` to execute the unit tests via
```

File: libs/fluid-elements/popover/jest.config.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
```
Comment (lukasholzer) [inline libs/fluid-elements/popover/jest.config.js:1]: this file can be removed and changed to `jest.config.json`

```json
{"name": "fluid-popover"}
```
That is needed to test with bazel

**Review by lukasholzer — CHANGES_REQUESTED** [review]

**Comment by lukasholzer** [comment]
@peter-affenzeller please resolve conflicts and fix unit tests

**Comment by peter-affenzeller** [comment]
@lukasholzer rebased again, but the tests are still failing on the CI for some reason, maybe you can help me out with this. locally all the tests are passing...

**Review by heartdisease — DISMISSED** [review]

**Review by lukasholzer — DISMISSED** [review]

**Review by bardosmisi — APPROVED** [review]
LGTM! 🦉

File: libs/fluid-elements/popover/src/lib/popover.ts [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+/**
+ * Set defaults for preventing overflow
+ * Applied to all instances of the FluidPopover
+ */
+preventOverflow.options = {
+  mainAxis: true,
+  altAxis: true,
+};
+
+type FluidPopoverOffsetFunction = ({
```
Comment (thomaspink) [inline libs/fluid-elements/popover/src/lib/popover.ts:46]: Should also be exported as it is part of the exported type `FluidPopoverOffset`

File: libs/fluid-elements/popover/src/lib/popover.ts [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+  @property({ attribute: false })
+  anchor: HTMLElement;
+
+  /** Defines the placement of the popper.js popover */
+  @property({ type: String, reflect: true })
+  placement: Placement = `bottom-start`;
+
+  /** Defines the placement of the popper.js popover */
+  @property({ type: Array, reflect: true })
+  get fallbackplacement(): Placement[] {
```
Comment (thomaspink) [inline libs/fluid-elements/popover/src/lib/popover.ts:132]: Not sure how this is for web-components, but normally we use camelCase for properties: `fallbackPlacement`

**Review by thomaspink — CHANGES_REQUESTED** [review]

**Comment by thomaspink** [comment]
I guess the issue for creating a "service" for overlays to be added on the body, to prevent overflow issues, should not be part of this PR?

File: libs/fluid-elements/popover/src/lib/popover.ts (hunk 7 above)
Comment (peter-affenzeller) [reply libs/fluid-elements/popover/src/lib/popover.ts:132]: I talked about this with @tomheller, who said that camelCase would be against the standard, referring to the [html attribute reference](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes).
I wanted to camelCase first too, but after the brief discussion I'm definitely on his side 😄

**Comment by peter-affenzeller** [comment]
@ffriedl89 mentioned in the discussion about the overflow issues that we should - at least for now - ignore this, see #1513

**Comment by ffriedl89** [comment]
@peter-affenzeller is this still something we have to deal with ourselves when using popper.js?

**Comment by peter-affenzeller** [comment]
@ffriedl89 yes, popper just positions the element, but does not modify the DOM.

**Review by thomaspink — APPROVED** [review]

**Comment by peter-affenzeller** [comment]
@ffriedl89 @lukasholzer I created a follow-up issue (#1645) for moving the popover to the end of the body.
the e2e-test failure is one of the radial chart tests, not related to the popover...

**Review by lukasholzer — APPROVED** [review]

**Review by heartdisease — DISMISSED** [review]

**Comment by lukasholzer** [comment]
@peter-affenzeller please reword and squash your commits to `chore(fluid-popover): Introduce fluid popover as web-component.`

We use only chore commits for the web components as we don't follow any semver and they are not part of our changelog.

**Review by heartdisease — APPROVED** [review]

**Comment by peter-affenzeller** [comment]
@lukasholzer done 🙂

**Review by lukasholzer — APPROVED** [review]

**Review by heartdisease — APPROVED** [review]

**Comment by lukasholzer** [comment]
manual cherry-picked into `8.x` and `8.0.x`

**Pull request merged**